components/PackageList.tsx

```diff
@@ -8,19 +9,28 @@ type PackageListProps = {
 }
 
 const PackageList: React.FC<PackageListProps> = (props) => {
+    const [packageList, setPackageList] = useState(props.items as { name: string, version: string }[]);
+
+    const onChange = (e: React.FormEvent<HTMLInputElement>) => {
+        const newValue = e.currentTarget.value;
+        setPackageList(props.items.filter(item => item.name.includes(newValue)));
+    }
```

Comment:
This can be written in a different syntax, again :). I think your version is completely fine, just showing alternatives:

```typescript
const onChange = (e: React.FormEvent<HTMLInputElement>) => 
  setPackageList(props.items.filter(item => item.name.includes(e.currentTarget.value))
```

Note the missing curly braces - this means that it's a shorthand for creating an arrow function that just has one return value and nothing else.

```typescript
// `a` and `b` are equivalent
const a = () => { return 1 };
const b = () => 1;
```

I have a personal preference in writing functions without the body as much as possible. I guess it helps me think in a functional way easier. But this is mostly a stylistic preference. 

Just giving perspective :)

Reply:
True `const newValue = e.currentTarget.value;` of course can be completely omitted.  
In fact I already know the shortcut to avoid using the {} when there is a single expression. That's a thing in Java too. Not sure why now I'm overlooking this syntax things a bit. It might be that I'm focusing  more on having the code running and doing what is supposed to do. A bit like wanting to understand the magic behind it. 
But yeah, I should also take care of this syntax things. 😄